prioritization_fees/src/account_priofees.rs

```diff
@@ -0,0 +1,220 @@
+                        acc_vec.push(value);
+                    }
+                    None => {
+                        accounts_by_prioritization_read_write.insert(*write_lock, vec![value]);
+                    }
+                }
+            }
+
+            for readlock in &transaction.readable_accounts {
+                match accounts_by_prioritization_read_write.get_mut(readlock) {
+                    Some(acc_vec) => {
+                        acc_vec.push(value);
+                    }
+                    None => {
+                        accounts_by_prioritization_read_write.insert(*readlock, vec![value]);
+                    }
+                }
+            }
+        }
+
+        let slot = produced_block.slot;
+        for (account, data) in accounts_by_prioritization_write {
+            let tx_count = data.len() as u64;
+            let cu_consumed = data.iter().map(|x| x.cu_consumed).sum();
+            let block_prio_data = BlockPrioData {
+                tx_prioritization: data,
+                nb_non_vote_tx: tx_count,
+                nb_total_tx: tx_count,
+                non_vote_cu_consumed: cu_consumed,
+                total_cu_consumed: cu_consumed,
+            };
+            match self.account_by_prio_fees_writeonly.get_mut(&account) {
+                Some(mut prio) => {
+                    prio.stats_by_slot.insert(slot, block_prio_data);
+                }
+                None => {
+                    let mut prio_fee = AccountPrio {
+                        stats_by_slot: BTreeMap::new(),
+                    };
+                    prio_fee.stats_by_slot.insert(slot, block_prio_data);
+                    self.account_by_prio_fees_writeonly
+                        .insert(account, prio_fee);
+                }
+            }
+        }
+
+        for (account, data) in accounts_by_prioritization_read_write {
+            let tx_count = data.len() as u64;
+            let cu_consumed = data.iter().map(|x| x.cu_consumed).sum();
+            let block_prio_data = BlockPrioData {
+                tx_prioritization: data,
+                nb_non_vote_tx: tx_count,
+                nb_total_tx: tx_count,
+                non_vote_cu_consumed: cu_consumed,
+                total_cu_consumed: cu_consumed,
+            };
+
+            match self.account_by_prio_fees_writeonly.get_mut(&account) {
+                Some(mut prio) => {
+                    prio.stats_by_slot.insert(slot, block_prio_data);
+                }
+                None => {
+                    let mut prio_fee = AccountPrio {
+                        stats_by_slot: BTreeMap::new(),
+                    };
+                    prio_fee.stats_by_slot.insert(slot, block_prio_data);
+                    self.account_by_prio_fees_writeonly
+                        .insert(account, prio_fee);
+                }
+            }
+        }
+
+        // cleanup old data
+        let last_slot = produced_block
```

Comment:
`last_slot` is very misleading (even worse: there is `self.last_slot` with completely different semantics from what I understand).

maybe call this one `min_slot_to_retain`